great_expectations/expectations/metrics/map_metric_provider.py

```diff
@@ -2585,7 +2562,7 @@ def _spark_multicolumn_map_condition_values(
 ):
     """Return values from the specified domain that match the map-style metric in the metrics dictionary."""
     (
-        boolean_mapped_unexpected_values,
+        unexpected_condition,
```

Comment:
@NathanFarmer Could we place keep this variable named `boolean_mapped_unexpected_values` as it was before (I was following the previous style in such methods across all execution engines and would like to keep it consistent, unless you have a strong reason for changing it now).  Thanks!

Reply:
The reason for changing this was that metrics are not returning booleans in all cases. Cases that return a window function would fail for the 1-line solution `filtered = df.filter(boolean_mapped_unexpected_values)`. The 2-line solution using `withColumn` creates a new boolean mapped column from this variable. You can see how I left it alone on line 2504 because it actually is a boolean mapping for all cases there.

Reply:
@NathanFarmer I cannot find all the line numbers involved (the UI here is confusing, but I follow your logic).  Thank you!  P.S.: Should we standardize all cases to use `unexpected_condition`?  Or do we want to preserve a special spot and variable name for the situations where it will always be strictly-boolean mapped?  Thoughts welcome.  Thank you.

Reply:
@NathanFarmer Perhaps we need to clean up our code?  Can you please help us out here -- I see in Pandas, SQL, and Spark the pattern that in some cases defines `unexpected_condition` and in other cases defines `boolean_mapped_unexpected_values` as variable names, but ultimately these variables are used the same way: in a `WHERE` type clause.  So are we simply misnaming this variable in one case for spark, because it is truly an (unexpected) condition and not merely boolean-mapped (unexpected) values?  If this is correct, then please go ahead and fix as you deem appropriate.  Thanks so much!

Reply:
@alexsherstinsky It is correct that the metric always returns an `unexpected_condition` for Spark. So for semantic correctness, I have changed the one remaining use of Spark `boolean_mapped_unexpected_values` to `unexpected_condition`. In most Spark cases `unexpected_condition` can be passed directly to a `WHERE` clause, except in the case of a window function. I went ahead and changed:
```
    data = df.filter(unexpected_condition)
```
to:
```
    data = df.withColumn("__unexpected", unexpected_condition)
    filtered = data.filter(F.col("__unexpected") == True).drop(F.col("__unexpected"))
```
for consistency in this single case.

Reply:
@NathanFarmer Thanks -- at least it is consistent, and we can revise later.  We now need to solve the remaining problem: the equivalency of results, when the sort order is not enforced.  If we can do this, then it is a big gain!